### PR TITLE
Customisable links support

### LINKED ISSUE #6663: Update Support Page
Changes required to the support page:

- Hide the button and text to add a subscription ID
- Remove the right-hand card 'Support'
- Move the 'Find out more ... or contact ..' link to the right hand area
- Ensure that the 'Get Support' box is only shown when commercial support is enabled

### LINKED ISSUE #6665: Change way supported version is detected
Currently we use the subscription ID setting.

Change this to use the metadata from the new `/rancherversion` endpoint - see: https://github.com/rancher/rancher/pull/38445/files

We should add a request to this API when the UI loads - before login and store that data. This will not change.

We can do this now and if we get a 404, we should populate empty data and store that the UI is not a supported deployment.

### PR DESCRIPTION
**Pull request by nwmac**
Fixes #6663 
Fixes #6665 

Supersedes #6736

### CODE REVIEW

**Review by richard-cox — CHANGES_REQUESTED** [review]
Tested a lot, including special deluxe mode, harvester standalone use case (CommunityLinks in support page) and some other weird edge cases. Apart from inline comments looks good

File: shell/config/home-links.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import { DOCS_BASE } from '@shell/config/private-label';
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/config/home-links.js:1]: Created https://github.com/rancher/dashboard/issues/7144 for this

File: shell/components/CommunityLinks.vue [hunk 2]
```diff
@@ -56,38 +44,41 @@ export default {
...
-            }, {});
-          }
-        } catch (e) {
-          console.error('Could not parse custom links setting', e); // eslint-disable-line no-console
-        }
+      // Combine the links
+      const all = [];
+
+      if (this.links.custom) {
+        all.push(...this.links.custom);
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/components/CommunityLinks.vue:71]: Not sure about this - created https://github.com/rancher/dashboard/issues/7145 to revist

File: shell/pages/c/_cluster/settings/links.vue [hunk 3]
```diff
@@ -55,64 +18,46 @@ export default {
...
-      value:              {},
+      value:              [],
       errors:             [],
-      showRestoredBanner: false
     };
   },
   computed: {
     ...mapGetters({ multiWithFallback: 'i18n/multiWithFallback' }),
 
+    // TODO: NEED TO SUPPOT MODRE FOR THE DEFAULT LINKS CONTROL
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/pages/c/_cluster/settings/links.vue]: The todo was done, so removed

**Review by richard-cox — APPROVED** [review]

**Pull request merged**